cli/mender-artifact/write.go

```diff
@@ -520,8 +520,8 @@ func getDeviceSnapshot(c *cli.Context) (string, error) {
 	// First echo to stdout such that we know when ssh connection is
 	// established (password prompt is written to /dev/tty directly,
 	// and hence impossible to detect).
-	args = append(args, "sudo", "-S", "/bin/sh", "-c",
-		`'echo "`+sshInitMagic+`" && mender snapshot dump | cat'`)
+	args = append(args, "/bin/sh", "-c",
+		`'echo "`+sshInitMagic+`" && [ $(id -u) -eq 0 ] || sudo_cmd="sudo -S"; $sudo_cmd mender snapshot dump | cat'`)
```

Comment:
@alfrunes can perhaps verify, but I think `sshInitMagic` needs to come after `sudo` because the client uses that string to identify the start of the payload data. If `sudo` comes after it, then there's a chance it will print `Password:`, which will then become part of your rootfs file instead of being displayed to the user.

Reply:
riight! got it! thanks